common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -256,6 +267,51 @@ def has_support(self, view, functionality):
             )
         return False
 
+    def validate(self):
+        """
+        Validates the xml for correct question separation. There must be only one inputtype in every <question>.
+        This is the override of the general XBlock method.
+        """
+
+        validation = super(CapaDescriptor, self).validate()
+        validation = StudioValidation.copy(validation)
+
+        # validate xml to check questions are correctly separated
+        is_valid_xml = self.validate_xml_for_inputtypes()
+
+        if not is_valid_xml:
+            validation.set_summary(
+                StudioValidationMessage(
+                    StudioValidationMessage.WARNING,
+                    u'Incorrect problem format detected!',
```

Comment:
Very cool, @muzaffaryousaf. Is the entire action-label linked to the action_link? 
I'd suggest the message, "The questions in this problem are not separated correctly."
For the action_label string, "Formatting Options".
The documentation URL for the action_link is 
http://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/course_components/create_problem.html#multiple-problems-in-one-component

Could you please add me to the staff for your course on Edge? I get a 403 Forbidden when I try to follow this link: https://studio.edge.edx.org/container/block-v1:AccessibilityX+AC100+2015+type@vertical+block@19cd93a2e3f24f0fa7d1ee4747029c50